resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -538,6 +538,9 @@ private[spark] class Client(
             if (!Utils.isLocalUri(jar)) {
               val path = getQualifiedLocalPath(Utils.resolveURI(jar), hadoopConf)
               val pathFs = FileSystem.get(path.toUri(), hadoopConf)
+              if (!pathFs.exists(path)) {
```

Comment:
Wait, no, this check is wrong. That's a glob, not a path.
Maybe check whether the globStatus result is null instead, to narrowly check what you are trying to?

Reply:
Oh that's a real mistake, thanks.